contributors/design-proposals/api-machinery/customresource-conversion-webhook.md

```diff
@@ -178,20 +178,27 @@ type ConversionRequest struct {
   UID types.UID
   // The version to convert given object to. E.g. "stable.example.com/v1"
   APIVersion string
-  // Object is the CRD object to be converted.
+  // Object is the CRD object or list of objects to be converted.
   Object runtime.RawExtension
+  // IsList indicates that this is a List operation and Object contains a list of items.
+  IsList bool
```

Comment:
Why is this necessary?

Another way to represent would be to have `List []runtime.RawExtension` and require that exactly one of `object` or `list` be set.

Reply:
List has its own meta data. Do we care let conversion webhook to modify that? right now the webhook is responsible for any change to that meta data (specifically changing the APIVersion). If we pass a list of objects, then the caller(APIServer) not the webhook is responsible for any list metadata changes.

Reply:
I don't see why the actual list object should be exposed to the conversion hook. Conversion is by list item by definition. @lavalamp's slice suggestion would make the interface obvious. We could even drop the `Object` field and just have an `Objects` instead.

Reply:
+1 for unifying the interface to something like `objects []runtime.RawExtension`, treating single and multi-item conversion requests consistently.